File: library/src/main/java/com/fanjun/keeplive/config/ForegroundNotification.java
package com.fanjun.keeplive.config;


import androidx.annotation.NonNull;

import java.io.Serializable;

/**
 * 默认前台服务样式
 */
public class ForegroundNotification implements Serializable {
    private String title;
    private String description;
    private int iconRes;
    private ForegroundNotificationClickListener foregroundNotificationClickListener;
    private ForegroundNotification(){

    }
    public ForegroundNotification(String title, String description, int iconRes, ForegroundNotificationClickListener foregroundNotificationClickListener) {
        this.title = title;
        this.description = description;
        this.iconRes = iconRes;
        this.foregroundNotificationClickListener = foregroundNotificationClickListener;
    }

    public ForegroundNotification(String title, String description, int iconRes) {
        this.title = title;
        this.description = description;
        this.iconRes = iconRes;
    }

    /**
     * 初始化
     * @return ForegroundNotification
     */
    public static ForegroundNotification ini(){
        return new ForegroundNotification();
    }
    /**
     * 设置标题
     * @param title 标题
     * @return ForegroundNotification
     */
    public ForegroundNotification title(@NonNull String title){
        this.title = title;
        return this;
    }
    /**
     * 设置副标题
     * @param description 副标题
     * @return ForegroundNotification
     */
    public ForegroundNotification description(@NonNull String description){
        this.description = description;
        return this;
    }
    /**
     * 设置图标
     * @param iconRes 图标
     * @return ForegroundNotification
     */
    public ForegroundNotification icon(@NonNull int iconRes){
        this.iconRes = iconRes;
        return this;
    }
    /**
     * 设置前台通知点击事件
     * @param foregroundNotificationClickListener 前台通知点击回调
     * @return ForegroundNotification
     */
    public ForegroundNotification foregroundNotificationClickListener(@NonNull ForegroundNotificationClickListener foregroundNotificationClickListener){
        this.foregroundNotificationClickListener = foregroundNotificationClickListener;
        return this;
    }

    public String getTitle() {
        return title==null?"":title;
    }

    public String getDescription() {
        return description==null?"":description;
    }

    public int getIconRes() {
        return iconRes;
    }

    public ForegroundNotificationClickListener getForegroundNotificationClickListener() {
        return foregroundNotificationClickListener;
    }
}

File: library/src/main/java/com/fanjun/keeplive/config/ForegroundNotification.kt
package com.fanjun.keeplive.config

import java.io.Serializable

/**
 * 默认前台服务通知样式配置
 * 支持构造器和链式调用两种初始化方式
 */
class ForegroundNotification : Serializable {

    var title: String = ""
        get() = field.ifEmpty { "" }
        private set

    var description: String = ""
        get() = field.ifEmpty { "" }
        private set

    var iconRes: Int = 0
        private set

    var foregroundNotificationClickListener: ForegroundNotificationClickListener? = null
        private set

    // 私有无参构造，用于链式调用
    private constructor()

    // 带点击事件的构造器
    constructor(
        title: String,
        description: String,
        iconRes: Int,
        foregroundNotificationClickListener: ForegroundNotificationClickListener?
    ) {
        this.title = title
        this.description = description
        this.iconRes = iconRes
        this.foregroundNotificationClickListener = foregroundNotificationClickListener
    }

    // 不带点击事件的构造器
    constructor(title: String, description: String, iconRes: Int) {
        this.title = title
        this.description = description
        this.iconRes = iconRes
    }

    // 链式调用方法
    fun title(title: String): ForegroundNotification = apply { this.title = title }
    fun description(description: String): ForegroundNotification = apply { this.description = description }
    fun icon(iconRes: Int): ForegroundNotification = apply { this.iconRes = iconRes }
    fun foregroundNotificationClickListener(listener: ForegroundNotificationClickListener): ForegroundNotification =
        apply { this.foregroundNotificationClickListener = listener }

    companion object {
        /**
         * 链式调用初始化入口
         */
        @JvmStatic
        fun ini(): ForegroundNotification = ForegroundNotification()
    }
}
